### PR TITLE
Download data endpoint

### PR DESCRIPTION
**Pull request by n0rdy**

### CODE REVIEW

File: xs2a-adapter-service-loader/src/main/java/de/adorsys/xs2a/adapter/service/loader/AdapterServiceLoader.java [hunk 1]
```diff
@@ -93,4 +94,13 @@ public Oauth2Service getOauth2Service(RequestHeaders requestHeaders) {
             .orElseThrow(() -> new AdapterNotFoundException(adapterId))
             .getOauth2Service(baseUrl, httpClientFactory, keyStore);
     }
+
+    public DownloadService getDownloadService(RequestHeaders requestHeaders) {
+        Aspsp aspsp = getAspsp(requestHeaders);
+        String adapterId = aspsp.getAdapterId();
+        String baseUrl = aspsp.getIdpUrl() != null ? aspsp.getIdpUrl() : aspsp.getUrl();
```
Comment (spetrychenko) [inline xs2a-adapter-service-loader/src/main/java/de/adorsys/xs2a/adapter/service/loader/AdapterServiceLoader.java]: could be replaced with Optional usage

File: xs2a-adapter-service-loader/src/test/java/de/adorsys/xs2a/adapter/service/impl/DownloadServiceImplTest.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    private static final String DOWNLOAD_URL = "https://www.example.com";
+    private static final RequestHeaders REQUEST_HEADERS = RequestHeaders.fromMap(Collections.emptyMap());
+    private static final ResponseHeaders RESPONSE_HEADERS = ResponseHeaders.fromMap(Collections.emptyMap());
+    private static final int HTTP_CODE_200 = 200;
+    private static final byte[] RESPONSE_BODY = "response body".getBytes();
+    private static final Response<byte[]> RESPONSE = new Response<>(HTTP_CODE_200, RESPONSE_BODY, RESPONSE_HEADERS);
+
+    @Mock
+    private AdapterServiceLoader adapterServiceLoader;
+    @Mock
```
Comment (spetrychenko) [inline xs2a-adapter-service-loader/src/test/java/de/adorsys/xs2a/adapter/service/impl/DownloadServiceImplTest.java]: empty line missed

File: xs2a-adapter-rest-impl/src/test/java/de/adorsys/xs2a/adapter/controller/DownloadControllerTest.java [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+    private static final int HTTP_CODE_200 = 200;
+    private static final byte[] RESPONSE_BODY = "response body".getBytes();
+    private static final ResponseHeaders RESPONSE_HEADERS = ResponseHeaders.emptyResponseHeaders();
+    private static final Response<byte[]> RESPONSE = new Response<>(HTTP_CODE_200, RESPONSE_BODY, RESPONSE_HEADERS);
+
+    private MockMvc mockMvc;
+
+    @Mock
+    private DownloadService downloadService;
+    @Mock
```
Comment (spetrychenko) [inline xs2a-adapter-rest-impl/src/test/java/de/adorsys/xs2a/adapter/controller/DownloadControllerTest.java]: missed empty line

File: xs2a-adapter-service-loader/src/main/java/de/adorsys/xs2a/adapter/service/loader/AdapterServiceLoader.java (hunk 1 above)
Comment (n0rdy) [reply xs2a-adapter-service-loader/src/main/java/de/adorsys/xs2a/adapter/service/loader/AdapterServiceLoader.java]: Done, thanks

File: xs2a-adapter-service-loader/src/test/java/de/adorsys/xs2a/adapter/service/impl/DownloadServiceImplTest.java (hunk 2 above)
Comment (n0rdy) [reply xs2a-adapter-service-loader/src/test/java/de/adorsys/xs2a/adapter/service/impl/DownloadServiceImplTest.java]: Done, thanks

File: xs2a-adapter-rest-impl/src/test/java/de/adorsys/xs2a/adapter/controller/DownloadControllerTest.java (hunk 3 above)
Comment (n0rdy) [reply xs2a-adapter-rest-impl/src/test/java/de/adorsys/xs2a/adapter/controller/DownloadControllerTest.java]: Done, thanks

**Pull request merged**